src/extension/sldService/src/test/java/org/geoserver/sldservice/rest/ClassifierTest.java

```diff
@@ -1851,4 +1861,39 @@ public void testCustomRampSingleValue() throws Exception {
         assertEquals("#00FF00", cm0.getColor().evaluate(null, String.class));
         assertEquals(1, cm0.getOpacity().evaluate(null, Double.class), 0);
     }
+
+    @Test
+    public void testNoDuplicatedRulesVectors() throws Exception {
+        final String restPath =
+                RestBaseController.ROOT_PATH
+                        + "/sldservice/cite:ClassificationLines/"
+                        + getServiceUrl()
+                        + ".xml?"
+                        + "attribute=cat2&ramp=CUSTOM&method=quantile&intervals=7&"
+                        + "colors=#FF071C,#CC0616,#82040E,#68030B,#530209,#420207,#350206";
+        Document dom = getAsDOM(restPath, 200);
+        print(dom);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        print(dom, baos);
+        String resultXml = baos.toString().replace("\r", "").replace("\n", "");
+        Rule[] rules =
+                checkSLD(resultXml.replace("<Rules>", sldPrefix).replace("</Rules>", sldPostfix));
+        assertTrue(rules.length < 7);
```

Comment:
Tests are not covering the changes and requirements fully. In particular:

- There should be three tests, covering open rules, closed rules, and explicit rules (the three modified methods in rules builder)
- Check that the rules are not just less, but the expected ones (check the threshold values), and also check colors associated (requirement, only the first colors in the provided list are meant to be used)

Reply:
Thanks for the review @aaime . feedback have been applied. Regarding explicit rules since I realized now that UniqueIntervalFunction was not affected by the duplicated problem I've removed the fix from corresponding method, but I've added anyway a test for that case.